extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcIdentityProvider.java

```diff
@@ -323,8 +324,14 @@ public TokenVerificationResult apply(TokenVerificationResult t) {
                 }
             }
 
-            return result;
+            tokenVerificationResult = result;
+        }
+
+        StepUpAuthenticationPolicy stepUpAuthPolicy = StepUpAuthenticationPolicy.getFromRequest(request);
```

Comment:
I propose for the moment to restrict it with `if (!isIdTokjen(token))`.  
Indeed, we can enforce the same claims values for ID tokens, but those error messages will be missed as Quarkus will just redirect the user to the OIDC provider...

I can look at making it work with the authorization code flow later

Reply:
I'd like to wait till I understand this comment https://github.com/quarkusio/quarkus/pull/47445#discussion_r2051537888 before I address this. I think users need to trust that when `@AuthenticationContext("myACR")` is on the endpoint, no request can access it unless the token has that claim value.

Maybe we should detect that it is unsupported scenario and throw exception (like illegal state) instead?

Reply:
@michalvavrik The situation with an ID token issued by the OIDC provider having no required acrs is indeed exceptional. But it is typically just `302` that we can do when it is `web-app`. Sure, let's keep it as is for now and tune later.

Reply:
I left it as is and added the warning. Comment if you want more.